### DISCUSSION TITLE
Export the privacy-preserving notification system into a library/sample project

### DISCUSSION THREAD

**Discussion by nicobao** [tutao/tutanota#6324] [General]
Hi,

I am Nicolas, co-founder and CTO of https://github.com/zkorum/zkorum - an open-source, verifiable and anonymous forum compatible with digital identity.

I am interested in using the privacy-preserving push notification system that you've been developing.

Any chance you can extract it from your product, and create a library or sample project?

I would be happy to co-maintain the repository.

Cheers!

[I originally created a [post](https://www.reddit.com/r/tutanota/comments/18muobi/export_privacypreserving_notification_system_into/) on Reddit because there were no issue template that seemed appropriate but then I was redirected to GH. I can't seem to remove the "bug" tag]

**Comment by charlag** [comment]
Hi
The notification system is basically SSE + Android service: https://github.com/tutao/tutanota/tree/master/app-android/app/src/main/java/de/tutao/tutanota/push

Alternatively I would recommend looking into [UnifiedPush](https://unifiedpush.org/)

**Reply by nicobao** [reply]
Thanks for your answer. My app is a PWA. I need to use the Web Push Notification API.
Do you know of any app that uses UnifiedPush behind a PWA?
